pkg/handler/common/provider/kubevirt.go

```diff
@@ -224,3 +228,22 @@ func KubeVirtStorageClassesWithClusterCredentialsEndpoint(ctx context.Context, u
 
 	return KubeVirtStorageClasses(ctx, kvKubeconfig)
 }
+
+func filterVMIPresets(vmiPresets apiv2.VirtualMachineInstancePresetList, quota kubermaticv1.MachineDeploymentVMResourceQuota) apiv2.VirtualMachineInstancePresetList {
+	filteredVMIPresets := apiv2.VirtualMachineInstancePresetList{}
+
+	// Range over the records and apply all the filters to each record.
+	// If the record passes all the filters, add it to the final slice.
+	for _, vmiPreset := range vmiPresets {
+		presetSpec := kubevirtv1.VirtualMachineInstancePresetSpec{}
+		if err := json.Unmarshal([]byte(vmiPreset.Spec), &presetSpec); err != nil {
+			return filteredVMIPresets
+		}
+		cpu := int(presetSpec.Domain.Resources.Requests.Cpu().AsApproximateFloat64())
```

Comment:
From what I understand from talking to @mfranczy , there are multiple ways to set the resource request for a VMIPreset, described here:
https://github.com/kubermatic/kubermatic/blob/master/pkg/ee/validation/machine/providers.go#L264-L273

So I think first you need to find where exactly its set, `presetSpec.Domain.Resources.Requests.` is not the only place